ngrep.c

```diff
@@ -1557,4 +1573,27 @@ char *win32_choosedevice(void) {
 }
 #endif
 
+/* Parse an unsigned long, exit program if anything goes wrong. */
+unsigned long parseulong(const char *s) {
+    unsigned long n = 0;
+    char *end = NULL;
 
+    errno = 0;
+    n = strtoul(s, &end, 0);
+
+    if (errno != 0 || *s == '\0' || end == NULL || *end != '\0') {
+        if (errno != 0) { /* GNU strtoul will set errno. */
+            perror("strtoul");
+        }
+        fprintf(stderr, "Could not convert unsigned long: `%s'.\n", s);
+
+        clean_exit(-1);
+    }
+
+    return n;
+}
+
+void dierange(char *msg, unsigned long value) {
+    fprintf(stderr, "`%s': value `%lu' is out of range.\n", msg, value);
```

Comment:
* Joshua Piccari <notifications@github.com>, 2015-08-21 23:10:56 Fri:
> > +void dierange(char *msg, unsigned long value) {
> > +    fprintf(stderr, "`%s': value `%lu' is out of range.\n", msg, value);
> If this routine expected a character array as the second parameter I
> think the feedback would be more valuable since in some cases we
> won't be able to output `value` properly as an `unsigned long`.

I believe the combination of `parseulong` (which errors out if the input is syntactically invalid) and `dierange` (which is called specifically for syntactically correct, but out of range integers) covers all cases that are relevant for this change set. I think that's what you are referring to in your edit.